### ISSUE TITLE
Mypy not catching class initialization/attribute access with literals:

### ISSUE THREAD

**Issue by quodlibetor** [python/mypy#2741]
This code doesn't catch the errors that I'd expect it to. In particular, none of the things in `main` error, but all the function calls give correct output:

```python
class Womp:
    def __init__(self, blah: int) -> None:
        self.blah = blah


def main():
    quux = Womp('a')  # no error
    bar.nope  # no error
    bar.blah + 'none'  # no error


def get_bad_womp(item: str) -> Womp:
    return Womp(item)  # correctly errors


def hello(a: Womp, b: str):
    a.blah + b  # correctly errors


def bloop(a: Womp):
    a.nope  # correctly errors
```

assuming that code is in `example.py` I get the following output:

```
$ mypy --fast-parser example.py
example.py:13: error: Argument 1 to "Womp" has incompatible type "str"; expected "int"
example.py:17: error: Unsupported operand types for + ("int" and "str")
example.py:21: error: "Womp" has no attribute "nope"
```

I couldn't find any open bugs that seemed to be about this, and I didn't see anything in the docs that suggested that immediately using class instances incorrectly wouldn't be caught.

```
$ mypy --version ; python --version
mypy 0.470
Python 3.5.2
```

**Comment by gvanrossum** [comment]
Your `main()` isn't being checked because you didn't add an annotation to it. Try
```python
def main() -> None:
    ...etc...
```

**Comment by gvanrossum** [comment]
Basically, read http://mypy.readthedocs.io/en/latest/basics.html#function-signatures .
